protocol/contracts/admin/VestingPayments.sol

```diff
@@ -0,0 +1,259 @@
+    function getVestingScheduleByAddressAndIndex(
+        address recipient,
+        uint256 index
+    ) external view override returns (VestingSchedule memory) {
+        return schedules[computeVestingScheduleIdForAddressAndIndex(recipient, index)];
+    }
+
+    /// @inheritdoc IVestingPayments
+    function getTotalVestedAtCurrentTime(bytes32 _vestingId) external view override returns (uint256) {
+        VestingSchedule memory _schedule = schedules[_vestingId];
+        return _getTotalVestedAtCurrentTime(_schedule);
+    }
+
+    /// @inheritdoc IVestingPayments
+    function getVestingSummary(
+        bytes32[] calldata _ids
+    ) external view override returns (VestingSummary[] memory summary) {
+        uint256 _length = _ids.length;
+        summary = new VestingSummary[](_length);
+        VestingSchedule memory _schedule;
+        for (uint256 i; i < _length; ++i) {
+            _schedule = schedules[_ids[i]];
+            summary[i] = VestingSummary(
+                _schedule.recipient,
+                _schedule.distributed,
+                _calculateTotalVestedAt(_schedule, uint40(block.timestamp))
+            );
+        }
+    }
+
+    /// @inheritdoc IVestingPayments
+    function getVestingIds() external view override returns (bytes32[] memory ids) {
+        ids = _activeVestingIds.values();
+    }
+
+    /// @inheritdoc IVestingPayments
+    function getReleasableAmount(bytes32 _vestingId) external view override returns (uint256) {
+        VestingSchedule storage _schedule = schedules[_vestingId];
+        return _calculateReleasableAmount(_schedule);
+    }
+
+    /// @inheritdoc IVestingPayments
+    function getTotalVestedAt(bytes32 _vestingId, uint40 _at) external view override returns (uint256) {
+        VestingSchedule memory _schedule = schedules[_vestingId];
+        return _calculateTotalVestedAt(_schedule, _at);
+    }
+
+    function _getTotalVestedAtCurrentTime(VestingSchedule memory _schedule) private view returns (uint256) {
+        return _calculateTotalVestedAt(_schedule, uint40(block.timestamp));
+    }
+
+    function _calculateReleasableAmount(
+          VestingSchedule storage _schedule
+    ) private view returns (uint256) {
+        // if account schedule is revoked, return the persisted checkpoint claimable vested delta. This also avoids an arithmetic underflow
+        return _schedule.revoked ? _schedule.revokedReleasable : _calculateTotalVestedAt(_schedule, uint40(block.timestamp)) - _schedule.distributed;
+    }
+
+    function _calculateTotalVestedAt(
+        VestingSchedule memory _schedule,
+        uint40 _releaseTime
+    ) private pure returns (uint256) {
+        if (_schedule.amount == 0) { return 0; }
+        // below cliff
+        if (_releaseTime <= _schedule.cliff) { return 0; }
+        // if revoked, return total vested at time of revoke (distributed + checkpointed delta)
+        if (_schedule.revoked) { return _schedule.revokedReleasable + _schedule.distributed; }
```

Comment:
Given this tweak to be `revokedReleasable + distributed`,  I think that means the line 233 doesn't need the ternary check now? It can just simplify to:

```
return _calculateTotalVestedAt(_schedule, uint40(block.timestamp)) - _schedule.distributed;
```
?